### PR TITLE
build(github): Add check for required migration

### PR DESCRIPTION
**Pull request by billyvg**
Add workflow for sentry python files that checks if a migration is needed.

### CODE REVIEW

**Comment by wedamija** [comment]
This looks good - can we provide some messaging explain what the user needs to do to fix this? (Just run `sentry django makemigrations -n <some_name>` and commit the generated files)

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+name: python
+on:
+  pull_request:
+    paths:
+      - 'src/sentry/**/*.py'
```
Comment (joshuarli) [inline .github/workflows/python.yml:5]:
```suggestion
      - 'src/**/*.py'
```

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        # setup python
+        - name: Set up Python ${{ steps.config.outputs.python-version }}
+          uses: actions/setup-python@v1
+          with:
+            python-version: ${{ steps.config.outputs.python-version}}
+
+        # setup pip
+        - name: Install pip
+          run: |
+            pip install --no-cache-dir --upgrade "pip>=20.0.2"
```
Comment (joshuarli) [inline .github/workflows/python.yml:67]:
```suggestion
            SENTRY_NO_VIRTUALENV_CREATION=1 make ensure-pinned-pip
```

Maybe not, put just putting it out there. I might want to rename `SENTRY_NO_VIRTUALENV_CREATION` before we do this, I didn't originally change it out of fear that some people might have dependend on that. Oh, or I could just add another env var `SKIP_ENSURE_VENV`, or even reuse some CI variables.

Comment (joshuarli) [reply .github/workflows/python.yml:67]: > or even reuse some CI variables.

@billyvg I can follow up with this sometime in the future, don't worry about it right now.

**Comment by billyvg** [comment]
@wedamija 
image

**Review by wedamija — APPROVED** [review]
🎉

**Review by joshuarli — APPROVED** [review]
Looks great.

**Pull request merged**